flocker/acceptance/endtoend/test_installer.py

```diff
@@ -0,0 +1,319 @@
+            COMPOSE_NODE0, 'rm', '-f'
+        )
+    )
+
+    d_node2_compose = remote_docker_compose(
+        test_case.client_ip, test_case.docker_host, COMPOSE_NODE1, 'stop')
+    d_node2_compose.addCallback(
+        lambda ignored: remote_docker_compose(
+            test_case.client_ip, test_case.docker_host, COMPOSE_NODE1,
+            'rm', '-f')
+    )
+
+    d = gather_deferreds([d_node1_compose, d_node2_compose])
+    d.addCallback(
+        lambda ignored: cluster.clean_nodes()
+    )
+
+    d.addCallback(
+        lambda ignored: delete_cloudformation_stack(test_case.stack_id)
+    )
+
+    return d
+
+
+def get_stack_report(stack_id):
+    output = check_output(
+        ['aws', '--region', REGION, 'cloudformation', 'describe-stacks',
+         '--stack-name', stack_id]
+    )
+    results = json.loads(output)
+    return results['Stacks'][0]
+
+
+def wait_for_stack_status(stack_id, target_status, time_limit=600):
+    start_time = time.time()
+    while True:
+        stack_report = get_stack_report(stack_id)
+        current_status = stack_report['StackStatus']
+        if current_status == target_status:
+            return stack_report
+        time_running = time.time() - start_time
+        if time_running > time_limit:
+            Message.new(
+                message='Timeout waiting for stack target status',
+                stack_id=stack_id,
+                target_status=target_status,
+                current_status=current_status,
+            ).write()
+            return False
+        else:
+            Message.new(
+                message='Waiting for stack target status',
+                stack_id=stack_id,
+                target_status=target_status,
+                current_status=current_status,
+                time_running=time_running,
+            ).write()
+
+            time.sleep(10)
+
+
+def create_cloudformation_stack():
+    """
+    """
+    # Request stack creation.
+    stack_name = CLOUDFORMATION_STACK_NAME + str(int(time.time()))
+    output = check_output(
+        ['aws', '--region', REGION, 'cloudformation', 'create-stack',
+         '--parameters', json.dumps(PARAMETERS),
+         '--stack-name', stack_name,
+         '--template-body', S3_CLOUDFORMATION_TEMPLATE]
+    )
+
+    output = json.loads(output)
+    stack_id = output['StackId']
+    Message.new(cloudformation_stack_id=stack_id)
+    stack_report = wait_for_stack_status(stack_id, 'CREATE_COMPLETE')
+    return stack_report
+
+
+def delete_cloudformation_stack(stack_id):
+    """
+    """
+    result = get_stack_report(stack_id)
+    outputs = result['Outputs']
+    s3_bucket_name = get_output(outputs, 'S3BucketName')
+    check_call(
+        ['aws', 's3', 'rb', 's3://{}'.format(s3_bucket_name), '--force']
+    )
+
+    check_call(
+        ['aws', '--region', REGION, 'cloudformation', 'delete-stack',
+         '--stack-name', stack_id]
+    )
+
+    return wait_for_stack_status(stack_id, 'DELETE_COMPLETE')
```

Comment:
:question: These could be combined into a `Fixture` object which could then be used with `TestCase.useFixture`, which would automatically add a cleanup. I've found the abstraction useful in the past, but am not sold enough on it to recommend it without reservation.